### ISSUE TITLE
Fix S1854 FP: Value used in finally should LiveIn for all try blocks

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#9440]
Should not raise in complex scenarios related to Roslyn LVA:

```C#
public void UsedInFinally()
{
            int value = 42; // Noncompliant FP
            try
            {
                SomethingThatCanThrow();
                value = 0;
            }
            finally
            {
                Use(value);
            }
}
```
